src/Avalonia.Base/Interactivity/RoutedPropertyChangedEventArgs.cs

```diff
@@ -0,0 +1,46 @@
+﻿namespace Avalonia.Interactivity
+{
+    /// <summary>
+    /// Provides both old and new property values with a routed event.
+    /// </summary>
+    /// <typeparam name="T">The type of values.</typeparam>
+    public class RoutedPropertyChangedEventArgs<T> : RoutedEventArgs
```

Comment:
Adding this new, generic EventArgs class to communicate new/old values is the only thing that I think can really be debated in this PR.

Reply:
Yes. It is a bit confusing and might cause questions why it's here, and not in PropertyChanged event, for example.

Reply:
But I wonder what's others opinion here. As I am not against it.

Reply:
Yea, when in doubt follow WPF/WinUI is generally best. I'm not sure what is best here. 

We could do what WinUI does though and have custom event args just for this. Then more information could be added in the future and it isnt duplicating the other property changed event information.

WPF is a bit strange here I suppose.

Reply:
I think WinUI gets it right here by having a specialized event args, we should probably follow that rather than WPF.

Reply:
PR https://github.com/AvaloniaUI/Avalonia/pull/11544 opened to switch the event args type